mdal/frmts/mdal_2dm.cpp

```diff
@@ -187,26 +193,30 @@ std::unique_ptr<MDAL::Mesh> MDAL::Driver2dm::load( const std::string &meshFile,
       face.resize( faceVertexCount );
 
       // chunks format here
-      // E** id vertex_id1, vertex_id2, ... material_id (elevation - optional)
+      // E** id vertex_id1, vertex_id2, vertex_id3, material_id [, aux_column_1, aux_column_2, ...]
       // vertex ids are numbered from 1
       // Right now we just store node IDs here - we will convert them to node indices afterwards
       assert( chunks.size() > faceVertexCount + 1 );
 
       for ( size_t i = 0; i < faceVertexCount; ++i )
         face[i] = MDAL::toSizeT( chunks[i + 2] ) - 1; // 2dm is numbered from 1
 
-      // OK, now find out if there is optional cell elevation (BASEMENT 3.x)
-      if ( chunks.size() == faceVertexCount + 4 )
-      {
+      // This assertion will fail if a mesh has fewer material ID columns than
+      // promised by the NUM_MATERIAL_PER_ELEM tag.
+      assert( chunks.size() - 5 >= materialCount );
```

Comment:
To support BASEMENT 3.x, I would do a logic a bit different here. 

`if chunks.size() - 5 = materialCount` -> current implementaion
`if chunks.size() - 5 = materialCount` -> take the one "extra" columen as `elevation - optional` (even NUM_MATERIALS_PER_ELEM==0)